contracts/WadMath.sol

```diff
@@ -76,19 +81,15 @@ library WadMath {
         }
     }
 
-    // Babylonian method, from Uniswap: https://github.com/Uniswap/uniswap-v2-core/blob/v1.0.1/contracts/libraries/Math.sol
-    // (via https://ethereum.stackexchange.com/a/87713/64318).
-    function wadSqrt(uint y) internal pure returns (uint z) {
-        y = y.mul(WAD);
-        if (y > 3) {
-            z = y;
-            uint x = y / 2 + 1;
-            while (x < z) {
-                z = x;
-                x = (y / x + x) / 2;
-            }
-        } else if (y != 0) {
-            z = 1;
+    function wadCbrt(uint y) internal pure returns (uint root) {
+        if (y > 0 ) {
+            y = y.mul(WAD_SQUARED);
+            uint newRoot = y.add(2) / 3;
+            do {
+                root = newRoot;
+                newRoot = (root + root + ((y / root) / root)) / 3;
+            } while (newRoot < root);
         }
+	//require(root ** 3 <= y && (root + 1) ** 3 > y);
```

Comment:
Would this be an `assert`, or something that you would like to test by fuzzing?

Reply:
Yeah I think something to test by fuzzing.  This finding the cube root thing turned into another rabbit hole...  I think what I have now is fine though, very short simple code and very cheap in gas.